com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/Templates/VFX/Config.template.hlsl

```diff
@@ -0,0 +1,296 @@
+$splice(VFXDefineSpace)
+
+$splice(VFXDefines)
+
+// Explicitly defined here for now (similar to how it was done in the previous VFX code-gen)
+#define HAS_ATTRIBUTES 1
+
+#define VFX_NEEDS_COLOR_INTERPOLATOR (VFX_USE_COLOR_CURRENT || VFX_USE_ALPHA_CURRENT)
+#if HAS_STRIPS
+#define VFX_OPTIONAL_INTERPOLATION
+#else//
+#define VFX_OPTIONAL_INTERPOLATION nointerpolation
+#endif
+
+ByteAddressBuffer attributeBuffer;
+
+#if VFX_HAS_INDIRECT_DRAW
+StructuredBuffer<uint> indirectBuffer;
+#endif
+
+#if USE_DEAD_LIST_COUNT
+ByteAddressBuffer deadListCount;
+#endif
+
+#if HAS_STRIPS
+Buffer<uint> stripDataBuffer;
+#endif
+
+// #if WRITE_MOTION_VECTOR_IN_FORWARD || USE_MOTION_VECTORS_PASS
+ByteAddressBuffer elementToVFXBufferPrevious;
+// #endif
+
+CBUFFER_START(outputParams)
+    float nbMax;
+    float systemSeed;
+CBUFFER_END
+
+// Helper macros to always use a valid instanceID
+#if defined(UNITY_STEREO_INSTANCING_ENABLED)
+    #define VFX_DECLARE_INSTANCE_ID     UNITY_VERTEX_INPUT_INSTANCE_ID
+    #define VFX_GET_INSTANCE_ID(i)      unity_InstanceID
+#else
+    #define VFX_DECLARE_INSTANCE_ID     uint instanceID : SV_InstanceID;
+    #define VFX_GET_INSTANCE_ID(i)      input.instanceID
+#endif
+
+#include "Packages/com.unity.render-pipelines.high-definition/Runtime/VFXGraph/Shaders/VFXCommon.hlsl"
+#include "Packages/com.unity.visualeffectgraph/Shaders/VFXCommon.hlsl"
+
+$splice(VFXParameterBuffer)
+
+$splice(VFXGeneratedBlockFunction)
+
+struct AttributesElement
+{
+    uint index;
+    Attributes attributes;
+#if HAS_STRIPS
+    uint relativeIndexInStrip;
+    StripData stripData;
+#endif
+};
+
+bool ShouldCull(uint index)
+{
+    uint deadCount = 0;
+#if USE_DEAD_LIST_COUNT
+    deadCount = deadListCount.Load(0);
+#endif
+    return (index >= asuint(nbMax) - deadCount);
+}
+
+float3 GetSize(Attributes attributes)
+{
+    float3 size3 = float3(attributes.size,attributes.size,attributes.size);
+
+    // TODO: Currently these do not get defined/generated.
+#if VFX_USE_SCALEX_CURRENT
+    size3.x *= attributes.scaleX;
+#endif
+#if VFX_USE_SCALEY_CURRENT
+    size3.y *= attributes.scaleY;
+#endif
+#if VFX_USE_SCALEZ_CURRENT
+    size3.z *= attributes.scaleZ;
+#endif
+
+#if HAS_STRIPS
+     // Add an epsilon so that size is never 0 for strips
+     size3 += size3 < 0.0f ? -VFX_EPSILON : VFX_EPSILON;
+#endif
+
+    return size3;
+}
+
+#if HAS_STRIPS
+#define PARTICLE_IN_EDGE (id & 1)
+float3 GetParticlePosition(uint index)
+{
+    struct Attributes attributes = (Attributes)0;
```

Comment:
Why this `struct` here ? Does it have any meaning in HLSL or is it a typo ?

Reply:
Good catch, this function was a simple copy/paste from the original template which contains the same thing:
https://github.com/Unity-Technologies/Graphics/blob/70d8490a694d9443549fc2619b5677238522d552/com.unity.visualeffectgraph/Shaders/VFXParticleCommon.template#L259

Reply:
We have Macro for zero initilaization that you should used: ZERO_INITIALIZE()